x-pack/plugin/text-structure/src/main/java/org/elasticsearch/xpack/textstructure/transport/TransportFindFieldStructureAction.java

```diff
@@ -58,21 +61,18 @@ protected void doExecute(Task task, FindFieldStructureAction.Request request, Ac
             .setFetchSource(true)
             .setQuery(QueryBuilders.existsQuery(request.getField()))
             .setFetchSource(new String[] { request.getField() }, null)
-            .execute(ActionListener.wrap(searchResponse -> {
-                long hitCount = searchResponse.getHits().getHits().length;
+            .execute(listener.delegateFailureAndWrap((delegate, searchResponse) -> {
+                var hitCount = searchResponse.getHits().getHits().length;
                 if (hitCount < AbstractFindStructureRequest.MIN_SAMPLE_LINE_COUNT) {
-                    listener.onFailure(
+                    delegate.onFailure(
                         new IllegalArgumentException("Input contained too few lines [" + hitCount + "] to obtain a meaningful sample")
                     );
                     return;
                 }
-                List<String> messages = getMessages(searchResponse, request.getField());
-                try {
-                    listener.onResponse(buildTextStructureResponse(messages, request));
-                } catch (Exception e) {
-                    listener.onFailure(e);
-                }
-            }, listener::onFailure));
+                var messages = getMessages(searchResponse, request.getField());
+                // As matching a regular expression might take a while, we run in a different thread to avoid blocking the network thread.
+                threadPool.generic().execute(ActionRunnable.supply(delegate, () -> buildTextStructureResponse(messages, request)));
```

Comment:
Why isn't `TextStructExecutor` used in this class (`TransportFindFieldStructureAction`) just like in the other two classes (`TransportFindMessageStructureAction`, `TransportFindStructureAction`)?

Reply:
I'm assuming we will want to leave this in when we eventually delete `TextStructExecutor`, whereas in the other two classes we will not.  Do you think this is correct? Otherwise, I can use `TextStructExecutor` here too and the future-dev can replace this line with `ActionListener.complete`.

Reply:
Good call, I think you're correct. LGTM!